tests/store/testdata.py

```diff
@@ -1,3 +1,5 @@
 the number one web server on the\nInternet.  It features support for HTTPS, virtual hosting, CGI, SSI,\nIPv6, easy scripting and database integration, request/response\nfiltering, many flexible authentication schemes, and more.\n", 'Subordinate': False, 'Provides': {'apache-website': {'Name': 'apache-website', 'Role': 'provider', 'Interface': 'apache-website', 'Optional': False, 'Limit': 0, 'Scope': 'container'}, 'local-monitors': {'Name': 'local-monitors', 'Role': 'provider', 'Interface': 'local-monitors', 'Optional': False, 'Limit': 0, 'Scope': 'container'}, 'logs': {'Name': 'logs', 'Role': 'provider', 'Interface': 'logs', 'Optional': False, 'Limit': 0, 'Scope': 'global'}, 'nrpe-external-master': {'Name': 'nrpe-external-master', 'Role': 'provider', 'Interface': 'nrpe-external-master', 'Optional': False, 'Limit': 0, 'Scope': 'container'}, 'website': {'Name': 'website', 'Role': 'provider', 'Interface': 'http', 'Optional': False, 'Limit': 0, 'Scope': 'global'}}, 'Requires': {'balancer': {'Name': 'balancer', 'Role': 'requirer', 'Interface': 'http', 'Optional': False, 'Limit': 1, 'Scope': 'global'}, 'logging': {'Name': 'logging', 'Role': 'requirer', 'Interface': 'syslog', 'Optional': False, 'Limit': 1, 'Scope': 'global'}, 'reverseproxy': {'Name': 'reverseproxy', 'Role': 'requirer', 'Interface': 'http', 'Optional': False, 'Limit': 1, 'Scope': 'global'}, 'vhost-config': {'Name': 'vhost-config', 'Role': 'requirer', 'Interface': 'apache-vhost-config', 'Optional': False, 'Limit': 1, 'Scope': 'global'}, 'website-cache': {'Name': 'website-cache', 'Role': 'requirer', 'Interface': 'http', 'Optional': False, 'Limit': 1, 'Scope': 'global'}}, 'Tags': ['app-servers'], 'min-juju-version': '0.0.0'}, 'owner': {'User': 'evarlast'}, 'promulgated': {'Promulgated': False}, 'supported-series': {'SupportedSeries': ['trusty']}}}, {'Id': 'cs:~nottrobin/trusty/apache2-wsgi-16', 'Meta': {'charm-metadata': {'Name': 'apache2-wsgi', 'Summary': 'Apache python app server, using mod_wsgi', 'Description': 'Install apache2 with mod_wsgi\nInstall python dependencies, from requirements.txt or from directory\nInstall ruby dependencies, from Gemfile or from directory\nExtract app files\nServe the app on ports 80 and 443\n', 'Subordinate': False, 'Provides': {'nrpe-external-master': {'Name': 'nrpe-external-master', 'Role': 'provider', 'Interface': 'nrpe-external-master', 'Optional': False, 'Limit': 0, 'Scope': 'container'}, 'web-server': {'Name': 'web-server', 'Role': 'provider', 'Interface': 'http', 'Optional': False, 'Limit': 0, 'Scope': 'global'}}, 'Requires': {'http': {'Name': 'http', 'Role': 'requirer', 'Interface': 'http', 'Optional': False, 'Limit': 1, 'Scope': 'global'}, 'mongodb': {'Name': 'mongodb', 'Role': 'requirer', 'Interface': 'mongodb', 'Optional': False, 'Limit': 1, 'Scope': 'global'}}, 'Categories': ['app-servers'], 'min-juju-version': '0.0.0'}, 'owner': {'User': 'nottrobin'}, 'promulgated': {'Promulgated': False}, 'supported-series': {'SupportedSeries': ['trusty']}}}]
```

Comment:
How do we update this test data? We should script it so that in the event we have to update it, like if the data structure changes we can do it quickly.

Reply:
I've made it so that you can do `./run exec npm run update-test-json` and it'll update the test JSON files.